docs/architecture/adr-111-nil-mempool.md

```diff
@@ -0,0 +1,239 @@
+internally:
+
+* it will reject any transaction being locally submitted or gossipped by a peer
+* when it is time to _reap_ (as it is currently called) the mempool the answer will always be empty
+* the application running on the proposer validator will add transactions it received
+  using the appchains's own mechanism via `PrepareProposal`.
+
+## Alternative Approaches
+
+These are the alternatives known to date:
+
+1. Keep the current model. Useful for basic apps, but clearly suboptimal for applications
+   with their own mechanism to disseminate transactions and particular performance requirements.
+2. Provide more efficient general-purpose mempool implementations.
+   This is an ongoing effort, but will take some time (and R&D effort) to come up with
+   advanced mechanisms -- likely highly configurable -- which then will have to be thoroughly tested.
+3. A similar approach to this one ([ADR110][adr-110]) whereby the application-specific
+   mechanism directly interacts with CometBFT via a newly defined gRPC interface.
+
+## Decision
+
+TBD
+
+## Detailed Design
+
+What this ADR proposes can already be achieved with an unmodified CometBFT since `v0.37.x`,
+albeit with a complex, poor UX.
+This core of this proposal is to make some internal changes so it is clear an simple for app developers,
+thus improving the UX.
+
+#### `nil` Mempool
+
+We propose a new mempool implementation, called `nil` Mempool, that effectively disables all mempool functionality
+within CometBFT.
+The `nil` Mempool implements the `Mempool` interface in a very simple manner:
+
+*	`CheckTx(tx types.Tx) (*abcicli.ReqRes, error)`: returns `nil, ErrNotAllowed`
+*	`RemoveTxByKey(txKey types.TxKey) error`: returns `ErrNotFound`
+* `ReapMaxBytesMaxGas(maxBytes, maxGas int64) types.Txs`: returns `nil`
+* `ReapMaxTxs(max int) types.Txs`: returns `nil`
+*	`Lock()`: does nothing
+* `Unlock()`: does nothing
+*	`Update(...) error`: returns `nil`
+* `FlushAppConn() error`: returns `nil`
+*	`Flush()`: does nothing
+* `TxsAvailable() <-chan struct{}`: returns a closed channel
+* `EnableTxsAvailable()`: does nothing
+* `SetTxRemovedCallback(cb func(types.TxKey))`: does nothing
+* `Size() int` returns 0
+* `SizeBytes() int64` returns 0
+
+Upon startup, the `nil` mempool reactor will advertise no channels to the peer to peer layer.
+
+### Configuration
+
+We propose the following changes to the `config.toml` file:
+
+```toml
+[mempool]
+# The type of mempool for this CometBFT node to use.
+#
+# Valid types of mempools supported by CometBFT:
+# - "flood" : clist mempool with flooding gossip protocol (default)
+# - "nil"   : nil-mempool (app needs an alternative tx dissemination mechanism)
+type = "nil"
+```
+
+The config validation logic will be modified to add a new rule that rejects a configuration file if:
+
+* the mempool is set to `nil`
+* `create_empty_blocks`, in `consensus` section, is set to `false`.
```

Comment:
I believe the reasoning here is because there is not a way to signify to comet to produce another block otherwise. Could we add a short sentence or two explaining this to avoid confusion for future readers.